README.md

```diff
@@ -69,35 +69,38 @@ $ git clone https://github.com/nvim-treesitter/nvim-treesitter.git
 
 ## Adding parsers
 
-Treesitter uses a different _parser_ for every language. It can be quite a pain to install, but fortunately `nvim-treesitter`
-provides two command to tackle this issue:
+Treesitter uses a different _parser_ for every language, which needs to be generated via `tree-sitter-cli` from a `grammar.js` file, then compiled to a `.so` library that needs to be placed in neovim's `runtimepath` (typically under `parser/{lang}.so`). To simplify this, `nvim-treesitter`
+provides commands to automate this process:
 
-- `TSInstall {language}` to install one or more parsers.
-  `TSInstall <tab>` will give you a list of supported languages, or select `all` to install them all.
-- `TSInstallInfo` to know which parser is installed.
+- `TSInstallInfo` to know which parsers are available and installed.
+- `TSInstall {language}` to install one or more parsers from a generated `c` file. (This requires a `C` compiler in your path.)
+- `TSInstallFromGrammar {language}` to install one or more parsers from the original `grammar.js`. (In addition to a `C` compiler, this requires the `tree-sitter-cli` executable in your path; see https://tree-sitter.github.io/tree-sitter/creating-parsers#installation for installation instructions.)
 - `TSUpdate` to update already installed parsers
```

Comment:
Could you also add what update means in that context (install the revision of `lockfile.json` if parser contained in this file or else newest master).

Reply:
With pleasure, if you explain in a bit more detail what it actually does ;)

Reply:
Ok, this is a bit more complicated... Probably that's the reason why it's not mentioned here.

- `:TSUpdate` is just `:TSInstall` without the `X is already installed. Reinstall?` query.
- The version that `:TSInstall` uses is described in `lockfile.json`. We pin the parsers because the parser revisions must be compatible with our queries.
- When you update, lockfile.json might change. TSUpdate reinstalls the new version in that case (thus #682). We recommend to run `:TSUpdate` when updating nvim-treesitter

See #682.

Reply:
How about this? I restructured the explanation a bit to avoid the forward reference. (It _is_ getting a bit long now, but I think explicit is good at this stage.)

Reply:
Yes, your explanation sounds good!